lanelet2/lanelet2_routing/src/Route.cpp

```diff
@@ -417,5 +424,21 @@ Route::Errors Route::checkValidity(bool throwOnError) const {
   return errors;
 }
 
+void Route::setEndPoint(lanelet::ConstPoint3d end_point)
+{
+  if (lanelet::geometry::distance(end_point.basicPoint2d(),shortestPath_.back().polygon2d()) > 1) // more than 1 meter away allowing room for error
```

Comment:
What does lanelet::geometry::distance return when the point is inside the polygon? Can we be sure it will be less than 1?